internal/liveness/metrics_util.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package liveness
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/ceph/ceph-csi/internal/util"
+	"k8s.io/klog"
+)
+
+const (
+	// volIDVersion is the version number of volume ID encoding scheme
+	volIDVersion uint16 = 1
+
+	// csiConfigFile is the location of the CSI config file
+	csiConfigFile = "/etc/ceph-csi-config/config.json"
+
+	// RADOS namespace to store CSI specific objects and keys
+	radosNamespace = "csi"
+)
+
+// CheckImageInCluster check if image in cluster
+func CheckImageInCluster(imageName string, imagePool string, monitors string, cr *util.Credentials) bool {
+
+	klog.V(4).Infof("rbd: check exist of image %s", imageName)
+	stdout, _, err := ExecCommand(
+		"rbd",
+		"-m", monitors,
+		"--id", cr.ID,
+		"--keyfile="+cr.KeyFile,
+		"-c", util.CephConfigPath,
+		"info", imagePool+"/"+imageName)
+	if err != nil {
+		klog.Errorf(fmt.Sprintf("failed to get image info, command output: %s", err.Error()))
+		return false
+	}
+	if strings.Contains(string(stdout), "No such file or directory") {
+		klog.V(4).Info("image not found in cluster")
+		return false
+	}
+	return true
+}
+
+// CheckSubVolumeInCluster check if subvolume in cluster
+func CheckSubVolumeInCluster(monitors string, subvolume string, volumegroup string, volume string, cr *util.Credentials) bool {
+
+	klog.V(4).Infof("cephfs: check exist of subvolume %s", subvolume)
+	stdout, _, err := ExecCommand(
+		"ceph",
+		"-m", monitors,
+		"--id", cr.ID,
+		"--keyfile="+cr.KeyFile,
+		"-c", util.CephConfigPath,
+		"fs", "subvolume", "ls", volume, volumegroup)
+	if err != nil {
+		klog.Errorf(fmt.Sprintf("failed to get subvolume list, command output: %s", err.Error()))
+		return false
+	}
+	if strings.Contains(string(stdout), subvolume) {
+		return true
+	}
+	klog.V(4).Info("subvolume not found in cluster")
+	return false
+}
+
+// ExecCommand seperate stdout with stderr
+func ExecCommand(program string, args ...string) (stdout, stderr []byte, err error) {
```

Comment:
we already have a function to run the command make use of it.

Reply:
done